packages/patternfly-4/react-core/src/components/Nav/Nav.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** NavList elements */
+  children(props, propName, componentName) {
+    const prop = props[propName];
+
+    let error = null;
+    React.Children.forEach(prop, child => {
+      if (child.type.componentType !== 'NavList') {
+        error = new Error(`${componentName} children should be of type NavList.`);
+      }
+    });
+    return error;
+  },
+  /** Additional classes added to the Nav */
+  className: PropTypes.string,
+  /** Non-visible label to describe the nav */
+  ariaLabel: PropTypes.string
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  ariaLabel: ''
+};
+
+const renderLists = (children, updateActive, activeGroup, activeItem) =>
+  React.Children.map(children, (childNode, index) =>
+    React.cloneElement(childNode, {
+      groupId: index,
+      updateActive,
+      activeGroup,
+      activeItem
+    })
+  );
+
+class Nav extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeItem: -1,
+      activeGroup: -1
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const { activeItem, activeGroup } = this.state;
+    if (activeItem === nextState.activeItem && activeGroup === nextState.activeGroup) {
+      return false;
+    }
+    return true;
+  }
+
+  // Called back from NavListItem
+  updateActive(groupId, itemId) {
+    this.setState({
+      activeItem: itemId,
+      activeGroup: groupId
+    });
+  }
+
+  render() {
+    const { children, className, ariaLabel, ...props } = this.props;
+
+    // Init the group and item states if unknown
+    let { activeItem, activeGroup } = this.state;
```

Comment:
I don't think that shis should be our responsibility. If user forgets to set `activeGroup` and `activeItem` via props he should either be notified about it or nothing special should happen. But let's not do something magical like selecting first group and first item.

For instance app navigation should not be affected if user enters config, such config is still part of application, but is not present in app navigation. These applications exists and we should have them in mind as well.